eclipse-collections-code-generator/src/main/resources/impl/set/immutable/immutablePrimitiveSetFactoryImpl.stg

```diff
@@ -16,6 +16,8 @@ package org.eclipse.collections.impl.set.immutable.primitive;
 import org.eclipse.collections.api.<name>Iterable;
 import org.eclipse.collections.api.factory.set.primitive.Immutable<name>SetFactory;
 import org.eclipse.collections.api.set.primitive.Immutable<name>Set;
+<if(!primitive.charOrShortOrByteOrFloatPrimitive)>import org.eclipse.collections.api.set.primitive.Mutable<name>Set;
```

Comment:
Needed to have this import so that I can do:    
 Mutable<name>Set newSet = <name>Sets.mutable.empty();

Reply:
The new method supports the 'if' in the template file. This new import enables the refactoring of the withAll methods. The withAll method isn't auto-generated in the factories for char/short/byte/float thus not needed. So unless you have the new import excluded, checkstyle fails with unused import error. Does this make sense?

Reply:
I understand what the if does in the template file, I just don't understand why these specific four types need to be excluded from the import.

Reply:
If the import is generated the build fails with unused import . This import helps with the impl of a specific with all method that isn't present for the given types because the template doesn't generate it.

Reply:
Note that for other places in the same file where we had to differentiate by type, we did not add methods to Primitive. So to @donraab's point, this stands out as inconsistent. Take a look at streamMethods in the same file for an example of how to customize by type without modifying Primitive.

Reply:
Txh for the comments - I've adopted the same approach as streamMethods and reverted the changes in Primitive. I'm not convinced this approach is cleaner as it adds a massive switch to govern the import generation that could have been replaced by just a single line if but I'll be guided by you.